test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -1828,23 +1829,35 @@ protected void skipTestWaitingForLuceneFix(org.apache.lucene.util.Version lucene
     }
 
     /**
-     * Get a deterministic SecureRandom SHA1PRNG instance seeded by deterministic LuceneTestCase.random().
+     * In non-FIPS mode, get a deterministic SecureRandom SHA1PRNG instance seeded by deterministic LuceneTestCase.random().
+     * In FIPS mode, get a non-deterministic SecureRandom DEFAULT/BCFIPS instance seeded by deterministic LuceneTestCase.random().
      * @return SecureRandom SHA1PRNG instance.
-     * @throws NoSuchAlgorithmException SHA1PRNG algorithm not found.
+     * @throws NoSuchAlgorithmException SHA1PRNG or DEFAULT algorithm not found.
+     * @throws NoSuchProviderException BCFIPS algorithm not found.
      */
-    public static SecureRandom secureRandom() throws NoSuchAlgorithmException {
+    public static SecureRandom secureRandom() throws NoSuchAlgorithmException, NoSuchProviderException {
         return secureRandom(randomByteArrayOfLength(32));
     }
 
     /**
-     * Get a deterministic SecureRandom SHA1PRNG instance seeded by the input value.
-     * @param seed Byte array to use for seeding the SecureRandom SHA1PRNG instance.
-     * @return SecureRandom SHA1PRNG instance.
-     * @throws NoSuchAlgorithmException SHA1PRNG algorithm not found.
+     * In non-FIPS mode, get a deterministic SecureRandom SHA1PRNG instance seeded by the input value.
+     * In FIPS mode, get a non-deterministic SecureRandom DEFAULT/BCFIPS instance seeded by the input value.
```

Comment:
I assume this means that in Fips mode,  if a test uses this method, the reproduction line won't work as expected ?  

If that is the case, can we add a log warning to the secureRandomFips method ?

Reply:
I added a warning log. I used SetOnce to only log it once.